docs/vc-http-api.yml

```diff
@@ -282,6 +282,18 @@ components:
           "domain": "example.com",
           "challenge": "d436f0c8-fbd9-4e48-bbb2-55fc5d0920a8",
         }
+    DeriveCredentialOptions:
+      type: object
+      additionalProperties: false
+      description: Options for specifying how the derived credential is created.
+      properties:
+        nonce:
+          type: string
+          description: A nonce provided by the holder of the credential to be included into the LinkedDataProof. For example 6e62f66e-67de-11eb-b490-ef3eeefa55f2
+      example:
+        {
+          "nonce": "d436f0c8-fbd9-4e48-bbb2-55fc5d0920a8",
```

Comment:
As we need to include the nonce bytes into the proof, perhaps we should specify that the nonce option is base64 encoded. @tplooker @OR13

Reply:
As a general rule these days, if you catch your self picking a base-encoding format, you're probably wrong. :)

A few things to keep in mind:

1. Multibase exists, if you want to base-encode something, we should use multibase for the value. 
2. Protocols can limit the type of base-encoding used (e.g. base58btc only). It's the easiest way to not get locked into a base encoding that doesn't work for some subset of the community.
2. The nonce value should never be optional and never be "default empty" in a protocol that is susceptible to replays. I think I saw something to this effect in a PR that flew through my inbox... just noting that it'll be a hard objection from us if folks want to default to empty nonces.

That said, the nonce value being encoded as pure bytes really doesn't matter... don't think we need to base-encode it... you just need enough randomness to prevent replays.

Reply:
@msporny 
> Multibase exists, if you want to base-encode something, we should use multibase for the value.

The context for mentioning base64 (or encoding, in general) comes from the BBS+ Signatures implementations and spec examples using it. As the output uses encoded bytes, I was curious if the input should use the same so they end up being the same.

I agree that multibase is good.

> The nonce value should never be optional and never be "default empty" in a protocol that is susceptible to replays.

Yup. In this case the output is a derivation of the VC (and is, itself a VC), but I would imagine the common intention is to use the derivation once. @OR13 has opened a PR on BBS+ Signatures spec regarding the MAY vs MUST nonce topic (https://github.com/w3c-ccg/ldp-bbs2020/pull/43).

> That said, the nonce value being encoded as pure bytes really doesn't matter

Note: the holder and the verifier both include the pure bytes into their proof computation. The output BBS+ LD Proof nonce property uses encoded bytes.

Example from [BBS+ Signatures spec](https://w3c-ccg.github.io/ldp-bbs2020/):
> "nonce": "6i3dTz5yFfWJ8zgsamuyZa4yAHPm75tUOOXddR6krCvCYk77sbCOuEVcdBCDd/l6tIY="

Of course, the derive service can transform from the input to bytes for the proof computation and do the output encoding. (and, as currently described in this PR, this is what would have to happen).

Reply:
@tplooker I agree with this thread, and am bothered by the base64 use in that suite :) we should move to multibase in the next version.